### DISCUSSION TITLE
External trigger signal

### DISCUSSION THREAD

**Discussion by halonguyen** [KillerInk/FreeDcam#149] [Ideas]
Hi there, i have an idea that can you adding a feature like this?
When we triggering the shutter button, at the same time another signal is sent to the type c port (may be a close switch or 5v signal or anythings else through some pins of type c port v.v...).
For the easiest is sending some sound signal and we can capture that signal through headset and use it for triggered things like high speed sync flash v.v...
Like the iblazr app: https://youtu.be/cL_0qmynglQ

For the advanced: we need two more parameters to adjust the delay (in ms maybe) one for the time from we press the shutter button to the actual exposure start and one for (also at that time start) to the time when the external signal send.
See picture below for more precisely.
![Screenshot_20210228_174142-01](https://user-images.githubusercontent.com/25997712/109415879-d30d3400-79ed-11eb-9099-3a167b9cf662.png)
Thank!

**Comment by KillerInk** [comment]
hi. in theory it should be possible. but you have to provide more technical informations.
myself dont have a external flash and no plans to buy one.
and it looks like that most ext flashs use different signals.
so there are different frequenzes and duration of the signal that dertermines the strength, temperature etc
im sure there are funny things possbile todo with

**Reply by halonguyen** [reply]
Thank you for your reply, don't worry about how is the signal processed, for the hardware part, i'll implement it later 😄. 
FYI: the external trigger signal, we only need an action act like a switch. Any thing like freqcy, time, connect v.v... I can control it manually through a flash trigger (and that why, for the precisely time sync, we need two delay parameters). We only need a simple fire signal and send it to the trigger. (Ofcourse I can convert the sound signal or any signal you want to export to a right signal we need, using hardware) 😁
P/s: For testing, I own many of flash lights either triggers so don't worry about it too. 😁
Explain: https://youtu.be/4TkhIHVprJc

**Reply by KillerInk** [reply]
added a some stuff. its using the connected audiodevice. so if nothing is connected its using the speakers.
the trigger signal must get extended i think. i just added the two byte array as sample.

https://github.com/KillerInk/FreeDcam/commit/dd81b8806bbae1af8b459b30085631e438c0c80b
